manifests/pulp.pp

```diff
@@ -129,6 +142,39 @@
     }
   }
 
+  if $expose_mongodb {
+    include certs::mongodb
+    include certs::mongodb_client
+
+    Class['certs::mongodb_client'] -> Class['pulp']
```

Comment:
```suggestion
    Class['certs::mongodb_client'] ~> Class['pulp']
```

I think this will fix your issue, @ehelms

Reply:
I thin kthat will only cover if the certificates get created or changed? But doesn't cover I feel the whole of the issue which is if `/etc/pulp/server.conf` gets updated, then `httpd` should get restarted to pick up the changes. 

@ekohl adding you in case you had any thoughts as to why I don't seem to be seeing the refreshing happen

Reply:
I suspect Pulp doesn't actually manage the vhost and thus doesn't refresh Apache. I'd try `Class['certs::mongodb_client'] ~> Class['pulp', 'apache::service']`. Both since `pulp` will still refresh the workers.